tests/nfs_sc_delayed_binding_test.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	mayav1alpha1 "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	deploy "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/apps/v1/deployment"
+	container "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/container"
+	pvc "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/persistentvolumeclaim"
+	pts "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/podtemplatespec"
+	volume "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/volume"
+	provisioner "github.com/openebs/dynamic-nfs-provisioner/provisioner"
+)
+
+/*
+ * This test will perform following steps:
+ * 1. Create NFS Storageclass with volumeBinding mode WaitForFirstConsumer
+ * 2. Create PVC with above storageclass
+ * 3. Create busybox deployment with above PVC
+ * 4. Delete busybox deployment
+ * 5. Delete PVC
+ * 6. Delete NFS Storageclass
+ */
+
+var _ = Describe("TEST WaitForFirstConsumer binding mode for NFS PV", func() {
+	var (
+		accessModes          = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany}
+		capacity             = "2Gi"
+		applicationNamespace = "nfs-tests-ns"
+
+		app           = "busybox-nfs"
+		pvcName       = "pvc-nfs"
+		label         = "demo=nfs-deployment"
+		labelselector = map[string]string{
+			"demo": "nfs-deployment",
+		}
+		appReplica = int32(2)
```

Comment:
Why we need two application replicas? Is it intentional?

Reply:
we haven't added multiple replica for deployment in test, so added it